File: python3/SquirrelSimulation.py
class Solution:
    def minDistance(self, height: int, width: int, tree: List[int], squirrel: List[int], nuts: List[List[int]]) -> int:
    
        def distance(a, b):
            return abs(a[0] - b[0]) + abs(a[1] - b[1])
        
        shortestPath = 0
        firstIndex = -1
        maxSaving = float("-inf")
        for index, nut in enumerate(nuts):
            nut2tree = distance(nut, tree)
            nut2squi = distance(nut, squirrel)
            if (nut2tree - nut2squi) > maxSaving:
                maxSaving = nut2tree - nut2squi
                firstIndex = index
        
        shortestPath += distance(squirrel, nuts[firstIndex])        
        shortestPath += distance(nuts[firstIndex], tree)
        
        for index, nut in enumerate(nuts):
            if (index != firstIndex):
                shortestPath += (distance(nut, tree) * 2)
                
        return shortestPath

File: c++/SquirrelSimulation.cpp
class Solution {
public:

    int distance(vector<int>& a, vector<int>& b) {
	    return (abs(a[0] - b[0]) + abs(a[1] - b[1]));
    }

    int minDistance(int height, int width, vector<int>& tree, vector<int>& squirrel, vector<vector<int>>& nuts) {
	    int maxSaving = INT_MIN;
            int nut2Tree = 0;
	    int nut2Squi = 0;
	    int output = 0;
	    int firstNut = 0;

	    for (int i = 0; i < nuts.size(); ++i) {
		nut2Tree = distance(nuts[i], tree);
    		nut2Squi = distance(nuts[i], squirrel);
		if ((nut2Tree - nut2Squi) > maxSaving) {
			maxSaving = nut2Tree - nut2Squi;
			firstNut = i;
		}
	    }

	    output += distance(squirrel, nuts[firstNut]);
	    output += distance(tree, nuts[firstNut]);

	    for (int i = 0; i < nuts.size(); ++i) {
		    if (i != firstNut) {
			    output += (distance(tree, nuts[i]) * 2);
		    }
	    }

	    return output;
    }
};
